backend/src/govsg/services/govsg-callback.service.ts

```diff
@@ -338,11 +338,16 @@ const parseUserMessageWebhook = async (
   clientId: WhatsAppApiClient
 ): Promise<void> => {
   const { wa_id: whatsappId } = body.contacts[0]
-  const { id: messageId, type } = body.messages[0]
+  const { id: messageId, type, timestamp } = body.messages[0]
+  const timestampAsDate = new Date(parseInt(timestamp, 10) * 1000) // convert to milliseconds
   if (type === WhatsappWebhookMessageType.button) {
     const message = body.messages[0] as WhatsAppWebhookButtonMessage
     if (message.button.text === 'Create passcode') {
       const passcodeCreationWamid = message.context.id
+      await GovsgVerification.update(
```

Comment:
Why not put this under `await GovsgVerification.findOne({` and make sure they are all done sequentially within 1 transaction?

Reply:
I tried to combine these into 1 query, but the problem I ran into was that `update` did not support `include: ...`.
As for why this is not a transaction, it is okay if the `update` is successful but the subsequent `findOne` fails because we are still correctly measuring the metric. (Since by this point the button click has already happened)

Reply:
Yo guys, actually there's an update query that returns the object post-update also, u should use that instead of having 2 separate queries here

Reply:
Got it, Kishen. Thank you for trying out my suggestion and letting me know what's up!!